File: ordenamiento_seleccion/seleccion.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
using namespace std;

int main() {
    srand(time(0));
    const int n = 10;
    int arr[n];

    // generar nums aleatorios
    for (int i = 0; i < n; i++) {
        arr[i] = rand() % 100 + 1;
    }

    cout << "Números random: ";
    for (int i = 0; i < n; i++) cout << arr[i] << " ";
    cout << endl;

    // ordenamiento de selección
    for (int i = 0; i < n - 1; i++) {
        int minIdx = i;
        for (int j = i + 1; j < n; j++) {
            if (arr[j] < arr[minIdx]) {
                minIdx = j;
            }
        }
        int temp = arr[i];
        arr[i] = arr[minIdx];
        arr[minIdx] = temp;
    }

    cout << "Números random ordenados: ";
    for (int i = 0; i < n; i++) cout << arr[i] << " ";
    return 0;
}

File: ordenamiento_seleccion/seleccion.java
import java.util.Random;

public class Selection {
    public static void main(String[] args) {
        Random rand = new Random();
        int[] arr = new int[10];

        // generar nums aleatorios
        for (int i = 0; i < arr.length; i++) {
            arr[i] = rand.nextInt(100) + 1;
        }

        System.out.print("Números random: ");
        for (int n : arr) System.out.print(n + " ");
        System.out.println();

        // ordenamiento de selección
        for (int i = 0; i < arr.length - 1; i++) {
            int minIdx = i;
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[j] < arr[minIdx]) {
                    minIdx = j;
                }
            }
            int temp = arr[i];
            arr[i] = arr[minIdx];
            arr[minIdx] = temp;
        }

        System.out.print("Números random ordenados: ");
        for (int n : arr) System.out.print(n + " ");
    }
}